Tests/SentryTests/Networking/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,256 @@
+        let interceptor = SentryHttpInterceptor(request: fixture.request, cachedResponse: nil, client: nil)
+        let session = interceptor.createSession()
+        XCTAssertNotNil(session)
+        XCTAssertTrue(session.delegate === interceptor)
+        XCTAssertEqual(session.configuration, URLSessionConfiguration.default)
+    }
+    
+    func testStartLoading() {
+        let now = Date()
+        fixture.dateProvider.setDate(date: now)
+        
+        let sut = fixture.getSut()
+        sut.startLoading()
+        let session = sut.session as? TestURLSession
+        
+        XCTAssertEqual(session?.lastDataTask?.resumeDate, now)
+    }
+    
+    func testStopLoading() {
+        let now = Date()
+        fixture.dateProvider.setDate(date: now)
+        
+        let sut = fixture.getSut()
+        sut.startLoading()
+        let session = sut.session as? TestURLSession
+        sut.stopLoading()
+        
+        XCTAssertEqual(session?.invalidateAndCancelDate, now)
+        XCTAssertNil(sut.session)
+    }
+    
+    func testTaskCompletionWithoutError() {
+        let sut = fixture.getSut()
+        let session = sut.createSession()
+        let client = sut.client as? TestProtocolClient
+        client?.testCallback = { method, params in
+            XCTAssertEqual(method, "urlProtocolDidFinishLoading:")
+            XCTAssertEqual(sut, params["urlProtocol"] as? URLProtocol)
+        }
+        sut.urlSession(session, task: URLSessionDataTaskMock(), didCompleteWithError: nil)
+    }
+    
+    func testTaskCompletionWithError() {
+        let sut = fixture.getSut()
+        let session = sut.createSession()
+        let client = sut.client as? TestProtocolClient
+        let someError = NSError(domain: "errorDomain", code: -1, userInfo: nil)
+        client?.testCallback = { method, params in
+            XCTAssertEqual(method, "urlProtocol:didFailWithError:")
+            XCTAssertEqual(sut, params["urlProtocol"] as? URLProtocol)
+            XCTAssertEqual(someError, params["didFailWithError"] as? NSError)
+        }
+        sut.urlSession(session, task: URLSessionDataTaskMock(), didCompleteWithError: someError)
+    }
+    
+    func testTaskDidReceiveResponse() {
+        let sut = fixture.getSut()
+        let session = sut.createSession()
+        let client = sut.client as? TestProtocolClient
+        let response = URLResponse()
+        client?.testCallback = { method, params in
+            XCTAssertEqual(method, "urlProtocol:didReceive:cacheStoragePolicy:")
+            XCTAssertEqual(sut, params["urlProtocol"] as? URLProtocol)
+            XCTAssertEqual(response, params["didReceive"] as? URLResponse)
+            XCTAssertEqual(URLCache.StoragePolicy.notAllowed, params["cacheStoragePolicy"] as? URLCache.StoragePolicy)
+        }
+        sut.urlSession(session, dataTask: URLSessionDataTaskMock(), didReceive: response) { disposition in
+            XCTAssertEqual(disposition, .allow)
+        }
```

Comment:
`m`: We need to make sure the callbacks are actually called. When I comment out some code in `SentryHttpInterceptor` the tests are not failing. This applies to all callbacks.
```suggestion

        var testCallbackCalled = false
        client?.testCallback = { method, params in
            testCallbackCalled = true
            XCTAssertEqual(method, "urlProtocol:didReceive:cacheStoragePolicy:")
            XCTAssertEqual(sut, params["urlProtocol"] as? URLProtocol)
            XCTAssertEqual(response, params["didReceive"] as? URLResponse)
            XCTAssertEqual(URLCache.StoragePolicy.notAllowed, params["cacheStoragePolicy"] as? URLCache.StoragePolicy)
        }
        
        var completionHandlerCalled = false
        sut.urlSession(session, dataTask: URLSessionDataTaskMock(), didReceive: response) { disposition in
            completionHandlerCalled = true
            XCTAssertEqual(disposition, .allow)
        }
        
        XCTAssertTrue(completionHandlerCalled)
        XCTAssertTrue(testCallbackCalled)
```